tests/decodeoutput.h

```diff
@@ -25,114 +26,218 @@
 #ifndef decodeoutput_h
 #define decodeoutput_h
 
-#include "VideoDecoderDefs.h"
-#include "VideoDecoderInterface.h"
-
-#include <stdio.h>
-#include <vector>
-#include <sstream>
+#include "vppinputoutput.h"
+#include "VideoPostProcessHost.h"
+#include "VideoCommonDefs.h"
 
 #if  __ENABLE_MD5__
 #include <openssl/md5.h>
 #endif
 
-using namespace YamiMediaCodec;
+#ifdef __ENABLE_X11__
+#include <X11/Xlib.h>
+#include <va/va_x11.h>
+#endif
+
+#ifdef __ENABLE_TESTS_GLES__
+#include "./egl/gles2_help.h"
+#include "egl/egl_util.h"
+#include "egl/egl_vaapi_image.h"
+#endif
+
+#include <vector>
+#include <string>
 
 class DecodeOutput
 {
 public:
-    static DecodeOutput* create(IVideoDecoder* decoder, int mode);
-    virtual bool setVideoSize(int width, int height);
-    virtual Decode_Status renderOneFrame(bool drain) = 0;
-    Decode_Status processOneFrame(bool drain);
-    uint32_t renderFrameCount() { return m_renderFrames; };
-
-    virtual ~DecodeOutput() {};
-
+    virtual ~DecodeOutput(){}
+    virtual bool init();
+    virtual bool output(SharedPtr<VideoFrame>& frame) = 0;
+    SharedPtr<NativeDisplay> nativeDisplay(){ return m_nativeDisplay;}
+    virtual void setVideoSize(int with, int height)
+    {
+        m_width = with;
+        m_height = height;
+    }
+    static DecodeOutput* create(int renderMode, uint32_t fourcc, const char* inputFile, const char* outputFile);
 protected:
-    DecodeOutput(IVideoDecoder* decoder);
-    IVideoDecoder* m_decoder;
     int m_width;
     int m_height;
-    uint32_t m_renderFrames;
-    VideoFrameRawData m_frame;
+    SharedPtr<VADisplay> m_vaDisplay;
+    SharedPtr<NativeDisplay> m_nativeDisplay;
 };
 
 class DecodeOutputNull: public DecodeOutput
 {
-friend DecodeOutput* DecodeOutput::create(IVideoDecoder* decoder, int mode);
 public:
-    virtual Decode_Status renderOneFrame(bool drain);
-    ~DecodeOutputNull() {};
+    DecodeOutputNull() {}
+    ~DecodeOutputNull() {}
+    bool init();
+    bool output(SharedPtr<VideoFrame>& frame);
+};
 
+
+class ColorConvert
+{
+public:
+    uint8_t* convert(SharedPtr<VideoFrame>& frame, SharedPtr<VADisplay> display, uint32_t pitches[3], uint32_t offsets[3])
```

Comment:
use `const`when passing shared pointers by reference, unless you plan to modify the shared pointer.